### PR TITLE
Add ability to mock meta providers

### LINKED ISSUE #65: Meta Mocks...
**Enhancement**

Becoming dependent on meta providers, which is a good thing, it is likely we need a concise and easy way to mock responses to a meta provider via the harness.

### PR DESCRIPTION
**Pull request by kitsonk**
**Type:** feature

The following has been addressed in the PR:

* [X] There is a related issue
* [X] All code matches the [style guide](https://github.com/dojo/meta/blob/master/STYLE.md)
* [X] Unit or Functional tests are included in the PR

**Description:**

This PR provides the ability to mock meta providers.

Allows for mocking of meta providers when testing a widget.  The method `.mockMeta()` takes two arguments, first the meta provider class and the second argument being a map of methods to mock on the provider when the harness widget is created.

For example, to provide mocked `Dimensions` for a widget:

```ts
const widget = harness.widget(MyWidget);

const rootDimensions = {
    offset: { height: 100, left: 100, top: 100, width: 100 },
    position: { bottom: 200, left: 100, right: 200, top: 100 },
    scroll: { height: 100, left: 100, top: 100, width: 100 },
    size: { width: 100, height: 100 }
};
const emptyDimensions = {
    offset: { height: 0, left: 0, top: 0, width: 0 },
    position: { bottom: 0, left: 0, right: 0, top: 0 },
    scroll: { height: 0, left: 0, top: 0, width: 0 },
    size: { width: 0, height: 0 }
};

const handle = widget.mockMeta(Dimensions, {
    get(key: string | number) {
        return key === 'root' ? rootDimensions : emptyDimensions;
    }
});
```

The handle returned from the `.mockMeta` function can be used to remove the mocks.

To facilitate usage under TypeScript, there is a special context type (`MetaMockContext`) that is exported from the `harness` module.  This is designed to represent the run-time context of the mock methods, which will allow access to `this.invalidate()` and `this.getNode()` which may be required to create an appropriate mock.

For example, to invalidate a widget via the meta provider mock:

```ts
import harness, { MetaMockContext } from '@dojo/text-extras/harness';

const widget = harness.widget(MyWidget);

widget.mockMeta(Dimensions, {
    get(this: MetaMockContext<Dimensions>, key: string | number) {
        this.invalidate();
        return {}
    }
})
```

Resolves #65

### CODE REVIEW

File: src/harness.ts [hunk 1]
```diff
@@ -167,18 +183,23 @@ class WidgetHarness<P extends WidgetProperties, W extends Constructor<WidgetBase
...
+	protected onElementCreated(element: HTMLElement, key: string) {
+		this._afterCreate(element);
 	}
 
 	/**
 	 * Called by a harnessed widget's render spy, allowing potential assertion of the render
 	 * @param actual The render, just after `afterRender`
 	 */
-	actualRender(actual: DNode) {
+	public actualRender(actual: DNode) {
```
Comment (bitpshr) [inline src/harness.ts:202]: Why does this have to be explicitly marked as `public`?

**Review by bitpshr — APPROVED** [review]
I like the resulting API.

File: src/harness.ts (hunk 1 above)
Comment (kitsonk) [reply src/harness.ts:202]: It is unnecessary, though we have been trying to be more explicit in the visibility operators.  I can remove it if it is a concern.  I forgot I did it.

**Review by smhigley — APPROVED** [review]
The API looks good. This will make Listbox tests much prettier :)

**Pull request merged**